File: chat-server-spring/src/main/java/com/netmarble/chat/domain/repository/AttachmentRepository.java
package com.netmarble.chat.domain.repository;

import com.netmarble.chat.domain.model.Attachment;

import java.util.Optional;

/**
 * Attachment 저장소 인터페이스 (DDD - 인프라 독립)
 */
public interface AttachmentRepository {

    Attachment save(Attachment attachment);

    Optional<Attachment> findByMessageId(Long messageId);

    void delete(Attachment attachment);
}

File: server/chat-server/src/domain/repository/attachment.repository.ts
import { Attachment } from '../model/attachment.js';

export const ATTACHMENT_REPOSITORY = Symbol('AttachmentRepository');

export abstract class AttachmentRepository {
  abstract save(attachment: Attachment): Promise<Attachment>;
  abstract findByMessageId(messageId: number): Promise<Attachment | null>;
  abstract findByMessageIds(messageIds: number[]): Promise<Map<number, Attachment>>;
  abstract delete(attachment: Attachment): Promise<void>;
}
